### ISSUE TITLE
Failed to pull image "gcr.io/k8s-minikube/storage-provisioner-s390x:v3"

### ISSUE THREAD

**Issue by prajaktachaudhari** [kubernetes/minikube#9253]
**Steps to reproduce the issue:** 
1. sudo -E minikube start --vm-driver=none

**Full output of `minikube start` command used, if not already included:**
```ubuntu@jk-slave2:~/work$ sudo -E minikube start --vm-driver=none
* minikube v1.13.0 on Ubuntu 18.04 (s390x)
* Using the none driver based on user configuration
* Starting control plane node minikube in cluster minikube
* Running on localhost (CPUs=0, Memory=32240MB, Disk=19788MB) ...
* OS release is Ubuntu 18.04.2 LTS
* Preparing Kubernetes v1.19.0 on Docker 18.06.3-ce ...
  - kubelet.resolv-conf=/run/systemd/resolve/resolv.conf
    > kubeadm.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubectl.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubelet.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubectl: 39.88 MiB / 39.88 MiB [---------------] 100.00% 74.28 MiB p/s 1s
    > kubeadm: 35.88 MiB / 35.88 MiB [---------------] 100.00% 41.24 MiB p/s 1s
    > kubelet: 100.97 MiB / 100.97 MiB [-------------] 100.00% 73.37 MiB p/s 1s
* Configuring local host environment ...
*
! The 'none' driver is designed for experts who need to integrate with an existing VM
* Most users should use the newer 'docker' driver instead, which does not require root!
* For more information, see: https://minikube.sigs.k8s.io/docs/reference/drivers/none/
*
! kubectl and minikube configuration will be stored in /home/ubuntu
! To use kubectl or minikube commands as your own user, you may need to relocate them. For example, to overwrite your own settings, run:
*
  - sudo mv /home/ubuntu/.kube /home/ubuntu/.minikube $HOME
  - sudo chown -R $USER $HOME/.kube $HOME/.minikube
*
* This can also be done automatically by setting the env var CHANGE_MINIKUBE_NONE_USER=true
* Verifying Kubernetes components...
* Enabled addons: default-storageclass, storage-provisioner
* Done! kubectl is now configured to use "minikube" by default
```
**Pod status :**
```
ubuntu@jk-slave2:~/work$ kubectl get pods --all-namespaces
NAMESPACE     NAME                                READY   STATUS             RESTARTS   AGE
kube-system   coredns-f9fd979d6-rn47l             1/1     Running            0          2m35s
kube-system   etcd-jk-slave2                      1/1     Running            0          2m35s
kube-system   kube-apiserver-jk-slave2            1/1     Running            0          2m35s
kube-system   kube-controller-manager-jk-slave2   1/1     Running            0          2m35s
kube-system   kube-proxy-gpndw                    1/1     Running            0          2m35s
kube-system   kube-scheduler-jk-slave2            1/1     Running            0          2m35s
kube-system   storage-provisioner                 0/1     ImagePullBackOff   0          2m41s
```
**Additional logs for  storage-provisioner**
```
Events:
  Type     Reason            Age                    From                Message
  ----     ------            ----                   ----                -------
  Warning  FailedScheduling  4m11s (x3 over 4m17s)  default-scheduler   0/1 nodes are available: 1 node(s) had taint {node.kubernetes.io/not-ready: }, that the pod didn't tolerate.
  Normal   Scheduled         4m7s                   default-scheduler   Successfully assigned kube-system/storage-provisioner to jk-slave2
  Normal   Pulling           2m33s (x4 over 4m7s)   kubelet, jk-slave2  Pulling image "gcr.io/k8s-minikube/storage-provisioner-s390x:v3"
  Warning  Failed            2m33s (x4 over 4m6s)   kubelet, jk-slave2  Failed to pull image "gcr.io/k8s-minikube/storage-provisioner-s390x:v3": rpc error: code = Unknown desc = Error response from daemon: manifest for gcr.io/k8s-minikube/storage-provisioner-s390x:v3 not found
  Warning  Failed            2m33s (x4 over 4m6s)   kubelet, jk-slave2  Error: ErrImagePull
  Warning  Failed            2m20s (x6 over 4m6s)   kubelet, jk-slave2  Error: ImagePullBackOff
  Normal   BackOff           2m5s (x7 over 4m6s)    kubelet, jk-slave2  Back-off pulling image "gcr.io/k8s-minikube/storage-provisioner-s390x:v3"
```
Additional findings:
`v3` tag is missing for `storage-provisioner` image. https://console.cloud.google.com/gcr/images/k8s-minikube/GLOBAL/storage-provisioner-s390x?gcrImageListsize=30

**Comment by sharifelgamal** [comment]
Yeah it looks like v3 of storage-provisioner (and v2 for that matter) was only released for amd architecture. That's a bug we need to fix in our release process.

**Comment by afbjorklund** [comment]
It is the same as #9060 (for another architecture). As a workaround, you can build it yourself: `make storage-provisioner-image`

**Comment by afbjorklund** [comment]
We should make it multi-arch: #9061

**Comment by prajaktachaudhari** [comment]
Thanks @sharifelgamal and @afbjorklund for prompt replies.

**Comment by afbjorklund** [comment]
Fixed in PR #9334

**Comment by Prabhav-Thali** [comment]
Hi @afbjorklund,

I have observed that PR [#9334](https://github.com/kubernetes/minikube/pull/9334) was merged before 1.14.0 release. However, only amd64 images are seen on [gcr.io](https://console.cloud.google.com/gcr/images/k8s-minikube/GLOBAL/storage-provisioner?gcrImageListsize=30). Could you please provide some info on this?

**Comment by sharifelgamal** [comment]
We made the changes to the code, but have yet to actually push out the new images. We'll get to it soon.

**Comment by u2bo** [comment]
@sharifelgamal some arm64 image available?

**Comment by afbjorklund** [comment]
> I have observed that PR [#9334](https://github.com/kubernetes/minikube/pull/9334) was merged before 1.14.0 release. However, only amd64 images are seen on [gcr.io](https://console.cloud.google.com/gcr/images/k8s-minikube/GLOBAL/storage-provisioner?gcrImageListsize=30). Could you please provide some info on this?

Probably some quirk in the release process, can't really test it myself (no credentials)

But the _theory_ was that it should build and push the images for "all" architectures:

```console
$ make -n push-storage-provisioner-manifest
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o out/storage-provisioner-amd64 -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-amd64:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=amd64 .
CGO_ENABLED=0 GOOS=linux GOARCH=arm go build -o out/storage-provisioner-arm -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-arm:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=arm .
CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -o out/storage-provisioner-arm64 -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-arm64:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=arm64 .
CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build -o out/storage-provisioner-ppc64le -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-ppc64le:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=ppc64le .
CGO_ENABLED=0 GOOS=linux GOARCH=s390x go build -o out/storage-provisioner-s390x -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-s390x:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=s390x .
docker login gcr.io/k8s-minikube
set -x; for arch in amd64 arm arm64 ppc64le s390x; do docker push gcr.io/k8s-minikube/storage-provisioner-${arch}:v3; done
docker manifest create --amend gcr.io/k8s-minikube/storage-provisioner:v3 make -n push-storage-provisioner-manifest
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o out/storage-provisioner-amd64 -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-amd64:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=amd64 .
CGO_ENABLED=0 GOOS=linux GOARCH=arm go build -o out/storage-provisioner-arm -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-arm:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=arm .
CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -o out/storage-provisioner-arm64 -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-arm64:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=arm64 .
CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build -o out/storage-provisioner-ppc64le -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-ppc64le:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=ppc64le .
CGO_ENABLED=0 GOOS=linux GOARCH=s390x go build -o out/storage-provisioner-s390x -ldflags="-X k8s.io/minikube/pkg/storage.version=v3 -s -w -extldflags '-static'" cmd/storage-provisioner/main.go
docker build -t gcr.io/k8s-minikube/storage-provisioner-s390x:v3 -f deploy/storage-provisioner/Dockerfile  --build-arg arch=s390x .
docker login gcr.io/k8s-minikube
set -x; for arch in amd64 arm arm64 ppc64le s390x; do docker push gcr.io/k8s-minikube/storage-provisioner-${arch}:v3; done
docker manifest create --amend gcr.io/k8s-minikube/storage-provisioner:v3 gcr.io/k8s-minikube/storage-provisioner-amd64:v3 gcr.io/k8s-minikube/storage-provisioner-arm:v3 gcr.io/k8s-minikube/storage-provisioner-arm64:v3 gcr.io/k8s-minikube/storage-provisioner-ppc64le:v3 gcr.io/k8s-minikube/storage-provisioner-s390x:v3
set -x; for arch in amd64 arm arm64 ppc64le s390x; do docker manifest annotate --arch ${arch} gcr.io/k8s-minikube/storage-provisioner:v3 gcr.io/k8s-minikube/storage-provisioner-${arch}:v3; done
docker manifest push gcr.io/k8s-minikube/storage-provisioner:v3
rm out/storage-provisioner-s390x out/storage-provisioner-ppc64le out/storage-provisioner-arm out/storage-provisioner-arm64
gcr.io/k8s-minikube/storage-provisioner-amd64:v3 gcr.io/k8s-minikube/storage-provisioner-arm:v3 gcr.io/k8s-minikube/storage-provisioner-arm64:v3 gcr.io/k8s-minikube/storage-provisioner-ppc64le:v3 gcr.io/k8s-minikube/storage-provisioner-s390x:v3
set -x; for arch in amd64 arm arm64 ppc64le s390x; do docker manifest annotate --arch ${arch} gcr.io/k8s-minikube/storage-provisioner:v3 gcr.io/k8s-minikube/storage-provisioner-${arch}:v3; done
docker manifest push gcr.io/k8s-minikube/storage-provisioner:v3
rm out/storage-provisioner-s390x out/storage-provisioner-ppc64le out/storage-provisioner-arm out/storage-provisioner-arm64
```

The architectures are hardcoded: `ALL_ARCH = amd64 arm arm64 ppc64le s390x`
